neuraxle/metaopt/auto_ml.py

```diff
@@ -626,40 +634,52 @@ def _fit_data_container(self, data_container: DataContainer, context: ExecutionC
 
         :return: self
         """
-        training_data_container, validation_data_container = self.validation_split_function(data_container)
+        validation_splits = self.validation_split_function(data_container)
 
         for trial_number in range(self.n_trial):
             self.print_func('\ntrial {}/{}'.format(trial_number + 1, self.n_trial))
-
             auto_ml_data = self._load_auto_ml_data(trial_number)
-            p = copy.deepcopy(self.pipeline)
 
             with self.hyperparams_repository.new_trial(auto_ml_data) as repo_trial:
-                try:
+
+                for training_data_container, validation_data_container in validation_splits:
+                    p = copy.deepcopy(self.pipeline)
                     p.update_hyperparams(repo_trial.hyperparams)
                     repo_trial.set_hyperparams(p.get_hyperparams())
 
-                    repo_trial = self.trainer.fit(
-                        p=p,
-                        train_data_container=training_data_container,
-                        validation_data_container=validation_data_container,
-                        trial=repo_trial,
-                        context=context
-                    )
-
-                    repo_trial.set_success()
-                    self.print_func('trial {}/{} score: {}'.format(trial_number + 1, self.n_trial, repo_trial.get_validation_scores()[-1]))
-                except Exception as error:
-                    track = traceback.format_exc()
-                    self.print_func(track)
-                    repo_trial.set_failed(error)
+                    with repo_trial.new_validation_split(p) as repo_trial_split:
+                        try:
+                            repo_trial_split = self.trainer.fit(
+                                trial=repo_trial_split,
+                                train_data_container=training_data_container,
+                                validation_data_container=validation_data_container,
+                                context=context
+                            )
+
+                            repo_trial_split.set_success()
+
+                            self.print_func('trial {}/{} split {}/{} score: {}'.format(
+                                trial_number + 1,
+                                self.n_trial,
+                                repo_trial_split.split_number + 1,
+                                len(training_data_container),
+                                repo_trial_split.get_validation_score()
+                            ))
+
+                        except Exception as error:
+                            track = traceback.format_exc()
```

Comment:
maybe add another extra print before printing to stack to give some context on why it failed. E.g.: print hyperparam space, and the trial number and trial split number.

Reply:
lots of times, it fails just with some values and it's good to know them to restrict space

Reply:
*EDIT: don't print the space here. I was wrong. Print the current hyperparameters to explain why it failed. Sorry

Reply:
colice